stream/src/public_key/handshake.rs

```diff
@@ -27,15 +27,48 @@ pub struct Info<P: PublicKey> {
 
     /// Timestamp of the handshake (in epoch milliseconds).
     timestamp: u64,
+    /// c.f. [`Self::with_tag_over`]
+    continuation_tag: Option<<Sha256 as Hasher>::Digest>,
 }
 
 impl<P: PublicKey> Info<P> {
-    /// Create a new hello.
+    /// Create a new info.
     pub fn new(recipient: P, ephemeral_public_key: x25519::PublicKey, timestamp: u64) -> Self {
         Self {
             recipient,
             ephemeral_public_key,
             timestamp,
+            continuation_tag: None,
+        }
+    }
+
+    /// Create a new info with a tag linking to previous data.
+    ///
+    /// For example, to allow for the info to also be linked to the hello message sent by the other peer,
+    /// or having this information attest to a MAC using a previous shared session key.
+    pub fn with_tag_over(
+        data: &[u8],
+        recipient: P,
+        ephemeral_public_key: x25519::PublicKey,
+        timestamp: u64,
+    ) -> Self {
+        Self {
+            recipient,
+            ephemeral_public_key,
+            timestamp,
+            continuation_tag: Some(Sha256::new().update(data).finalize()),
```

Comment:
We already have a function that provides something like this (`hash`) but it isn't defined on the hasher trait.

I'd recommend moving `hash` to be defined like @BrendanChou suggested and then calling `Sha256::hash()`)?

```
    /// Convenience function for hashing a single message.
    fn hash(data: &[u8]) -> Self::Digest {
        let mut hasher = Self::new();
        hasher.update(data);
        hasher.finalize()
    }
    ```